bitcoin/src/crypto/key.rs

```diff
@@ -1394,6 +1394,22 @@ mod tests {
         let res = PublicKey::from_str(s);
         assert!(res.is_err());
         assert_eq!(res.unwrap_err(), ParsePublicKeyError::InvalidHexLength(8));
+    }
+
+    #[test]
+    fn public_key_from_str_invalid_str() {
+        // Ensuring test cases fail when PublicKey::from_str is used on invalid keys
+        let s = "042e58afe51f9ed8ad3cc7897f634d881fdbe49a81564629ded8156bebd2ffd1af191923a2964c177f5b5923ae500fca49e99492d534aa3759d6b25a8bc971b142";
+        assert_eq!(s.len(), 130);
+        let res = PublicKey::from_str(s);
+        assert!(res.is_err());
+        assert_eq!(res.unwrap_err(), ParsePublicKeyError::Encoding(FromSliceError::Secp256k1(secp256k1::Error::InvalidPublicKey)));
+
+        let s = "069e58afe51f9ed8ad3cc7897f634d881fdbe49a81564629ded8156bebd2ffd1af191923a2964c177f5b5923ae500fca49e99492d534aa3759d6b25a8bc971b133";
```

Comment:
can you also add tests for compressed keys?

Reply:
Added one for `secp256k1::Error::InvalidPublicKey`. Do let me know if more tests are needed.

Reply:
@sanket1729, added few more tests. Can you please take a look at the commented test. I think it should give `FromSliceError::InvalidKeyPrefix` but it is giving `FromSliceError::Secp256k1(secp256k1::Error::InvalidPublicKey))`. Though it is not wrong, but I suppose the former should be detected.

Reply:
@Sh0g0-1758 , that is related to hybrid keys. I think the those are best left ignored and forgotten.